tests/zkevm/test_worst_compute.py

```diff
@@ -282,3 +283,135 @@ def code_loop_precompile_call(calldata: Bytecode, attack_block: Bytecode):
         raise ValueError(f"Code size {len(code)} exceeds maximum code size {MAX_CODE_SIZE}")
 
     return code
+
+
+@pytest.mark.zkevm
+@pytest.mark.valid_from("Cancun")
+@pytest.mark.slow
+@pytest.mark.parametrize(
+    "gas_limit",
+    [36_000_000],
+)
+def test_worst_jumps(
+    blockchain_test: BlockchainTestFiller,
+    pre: Alloc,
+    fork: Fork,
+    gas_limit: int,
+):
+    """Test running a JUMP-intensive contract."""
+    env = Environment(gas_limit=gas_limit)
+    gsc = fork.gas_costs()
+
+    def jump_seq(i):
+        return Op.PUSH2(4 + 5 * i) + Op.JUMP + Op.JUMPDEST
+
+    gas_per_seq = gsc.G_VERY_LOW + gsc.G_MID + gsc.G_JUMPDEST
+    bytes_per_seq = 5
+    seqs_per_call = MAX_CODE_SIZE // bytes_per_seq
+    gas_per_call = gas_per_seq * seqs_per_call
+
+    # Create and deploy the jump-intensive contract
+    jumps_code = sum(jump_seq(i) for i in range(seqs_per_call))
+    jumps_address = pre.deploy_contract(code=jumps_code)
+
+    # Determine how many calls to the jumps contract to make.
```

Comment:
Ah, I usually have a `While` without a condition (that's supported), and let the tx run until it runs out of gas.

So, no need to calculate the exact limit since the prover will have to prove everything anyway. :)

Maybe you want to try it and see if you simplify it that way, you should get the same cycles (or even a tiny more).

Reply:
Right, this makes sense--the prover needs to prove the full tx even to show it reverted. I don't know much about how these clients work so I was a little worried that perhaps are some shortcuts for identifying if a tx runs out of gas, but I see by manually checking the cycle counts are similar that there is no issue there, so I just max out gas as you suggest.